server/base/src/main/java/org/apache/accumulo/server/ServerContext.java

```diff
@@ -279,4 +300,137 @@ public Ample getAmple() {
   public Set<String> getRecoveryDirs() {
     return serverConstants.getRecoveryDirs();
   }
+
+  /**
+   * Check to see if this version of Accumulo can run against or upgrade the passed in data version.
+   */
+  public static void ensureDataVersionCompatible(int dataVersion) {
+    if (!(ServerConstants.CAN_RUN.contains(dataVersion))) {
+      throw new IllegalStateException("This version of accumulo (" + Constants.VERSION
+          + ") is not compatible with files stored using data version " + dataVersion);
+    }
+  }
+
+  public void waitForZookeeperAndHdfs() {
+    log.info("Attempting to talk to zookeeper");
+    while (true) {
```

Comment:
This might be clearer if the try was lifted above the while and a local boolean used to terminate the while.  The code is fine as is - I just don't use break much and it gives me a pause when I see it - more of a personal preference.

Reply:
I agree. I looked at this and I am not even sure if getChildren() will return null or empty list so I am going to just leave it.